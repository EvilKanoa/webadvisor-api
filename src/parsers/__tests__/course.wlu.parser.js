const path = require('path');
const util = require('util');
const readFile = util.promisify(require('fs').readFile);
const { parseXMLCourse } = require('../course.wlu.parser');

describe('course.wlu.parser', () => {
  it('handles empty inputs', async () => {
    expect(await parseXMLCourse('')).toBeFalsy();
    expect(await parseXMLCourse(null)).toBeFalsy();
    expect(await parseXMLCourse(false)).toBeFalsy();
  });

  it.each([
    ['course with many sections', 'course.wlu.data.1.xml'],
    ['course with few sections', 'course.wlu.data.2.xml'],
    ['course with no sections', 'course.wlu.data.3.xml'],
  ])('parses a %s correctly', async (desc, filename) => {
    const xml = await readFile(path.join(__dirname, '__data__', filename));
    const result = await parseXMLCourse(xml);
    const str = util.inspect(result, {
      depth: 10,
      maxArrayLength: Infinity,
      sorted: true,
    });
    expect(str).toMatchSnapshot();
  });
});
