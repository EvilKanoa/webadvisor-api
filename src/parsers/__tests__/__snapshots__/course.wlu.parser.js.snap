// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`course.wlu.parser parses a course with few sections correctly 1`] = `
"{
  code: 'CP-386',
  credits: 0.5,
  description: 'Topics include operating system services, file systems, CPU scheduling, memory management, virtual memory, disk scheduling, deadlocks, concurrent processes, protection and distributed systems.Prerequisites: CP216 and CP264 (or CP217).',
  location: 'Waterloo',
  name: 'Operating Systems',
  sections: [
    {
      available: 200,
      capacity: 200,
      id: '933',
      meetings: [
        {
          available: 200,
          capacity: 200,
          day: 'monday',
          end: '18:50',
          id: '933',
          location: 'BA101',
          name: 'Lec A',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 200,
          capacity: 200,
          day: 'wednesday',
          end: '18:50',
          id: '933',
          location: 'BA101',
          name: 'Lec A',
          start: '17:30',
          type: 'LEC'
        }
      ]
    }
  ]
}"
`;

exports[`course.wlu.parser parses a course with many sections correctly 1`] = `
"{
  code: 'CP-104',
  credits: 0.5,
  description: 'An introductory course designed to familiarize the student with modern software developmenttechniques. Emphasis is on problem-solving and structured program design methodologies.Programming projects are implemented in a widely used high-level language.Registration Notes:  Must register for both a lecture and a lab.',
  location: 'Waterloo, Brantford',
  name: 'Intro To Programming',
  sections: [
    {
      available: 0,
      capacity: 52,
      id: '60 (28)',
      meetings: [
        {
          available: 166,
          capacity: 375,
          day: 'monday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'wednesday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'friday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 0,
          capacity: 52,
          day: 'monday',
          end: '16:50',
          id: '28',
          location: 'BA113',
          name: 'Lab L1',
          start: '14:30',
          type: 'LAB'
        }
      ]
    },
    {
      available: 19,
      capacity: 52,
      id: '60 (2382)',
      meetings: [
        {
          available: 166,
          capacity: 375,
          day: 'monday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'wednesday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'friday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 19,
          capacity: 52,
          day: 'thursday',
          end: '18:20',
          id: '2382',
          location: 'BA113',
          name: 'Lab L10',
          start: '16:0',
          type: 'LAB'
        }
      ]
    },
    {
      available: 38,
      capacity: 52,
      id: '60 (2521)',
      meetings: [
        {
          available: 166,
          capacity: 375,
          day: 'monday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'wednesday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'friday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 38,
          capacity: 52,
          day: 'friday',
          end: '10:50',
          id: '2521',
          location: 'BA113',
          name: 'Lab L11',
          start: '8:30',
          type: 'LAB'
        }
      ]
    },
    {
      available: 40,
      capacity: 52,
      id: '60 (2522)',
      meetings: [
        {
          available: 166,
          capacity: 375,
          day: 'monday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'wednesday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'friday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 40,
          capacity: 52,
          day: 'friday',
          end: '14:20',
          id: '2522',
          location: 'BA113',
          name: 'Lab L12',
          start: '12:0',
          type: 'LAB'
        }
      ]
    },
    {
      available: 30,
      capacity: 52,
      id: '60 (3688)',
      meetings: [
        {
          available: 166,
          capacity: 375,
          day: 'monday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'wednesday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'friday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 30,
          capacity: 52,
          day: 'monday',
          end: '19:20',
          id: '3688',
          location: 'BA113',
          name: 'Lab L13',
          start: '17:0',
          type: 'LAB'
        }
      ]
    },
    {
      available: 7,
      capacity: 30,
      id: '60 (3728)',
      meetings: [
        {
          available: 166,
          capacity: 375,
          day: 'monday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'wednesday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'friday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 7,
          capacity: 30,
          day: 'tuesday',
          end: '21:20',
          id: '3728',
          location: 'N1055',
          name: 'Lab L14',
          start: '19:0',
          type: 'LAB'
        }
      ]
    },
    {
      available: 0,
      capacity: 52,
      id: '60 (30)',
      meetings: [
        {
          available: 166,
          capacity: 375,
          day: 'monday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'wednesday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'friday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 0,
          capacity: 52,
          day: 'tuesday',
          end: '15:20',
          id: '30',
          location: 'BA113',
          name: 'Lab L2',
          start: '13:0',
          type: 'LAB'
        }
      ]
    },
    {
      available: 41,
      capacity: 52,
      id: '60 (31)',
      meetings: [
        {
          available: 166,
          capacity: 375,
          day: 'monday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'wednesday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'friday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 41,
          capacity: 52,
          day: 'tuesday',
          end: '17:50',
          id: '31',
          location: 'BA113',
          name: 'Lab L3',
          start: '15:30',
          type: 'LAB'
        }
      ]
    },
    {
      available: 51,
      capacity: 52,
      id: '60 (32)',
      meetings: [
        {
          available: 166,
          capacity: 375,
          day: 'monday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'wednesday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'friday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 51,
          capacity: 52,
          day: 'wednesday',
          end: '14:50',
          id: '32',
          location: 'BA113',
          name: 'Lab L4',
          start: '12:30',
          type: 'LAB'
        }
      ]
    },
    {
      available: 0,
      capacity: 52,
      id: '60 (389)',
      meetings: [
        {
          available: 166,
          capacity: 375,
          day: 'monday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'wednesday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'friday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 0,
          capacity: 52,
          day: 'wednesday',
          end: '17:20',
          id: '389',
          location: 'BA113',
          name: 'Lab L5',
          start: '15:0',
          type: 'LAB'
        }
      ]
    },
    {
      available: 39,
      capacity: 52,
      id: '60 (390)',
      meetings: [
        {
          available: 166,
          capacity: 375,
          day: 'monday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'wednesday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'friday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 39,
          capacity: 52,
          day: 'wednesday',
          end: '19:50',
          id: '390',
          location: 'BA113',
          name: 'Lab L6',
          start: '17:30',
          type: 'LAB'
        }
      ]
    },
    {
      available: 42,
      capacity: 52,
      id: '60 (1530)',
      meetings: [
        {
          available: 166,
          capacity: 375,
          day: 'monday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'wednesday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'friday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 42,
          capacity: 52,
          day: 'thursday',
          end: '10:50',
          id: '1530',
          location: 'BA113',
          name: 'Lab L7',
          start: '8:30',
          type: 'LAB'
        }
      ]
    },
    { available: undefined, capacity: undefined, id: '', meetings: [] },
    {
      available: 14,
      capacity: 52,
      id: '60 (2381)',
      meetings: [
        {
          available: 166,
          capacity: 375,
          day: 'monday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'wednesday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 166,
          capacity: 375,
          day: 'friday',
          end: '14:20',
          id: '60',
          location: 'BA201',
          name: 'Lec A',
          start: '13:30',
          type: 'LEC'
        },
        {
          available: 14,
          capacity: 52,
          day: 'thursday',
          end: '15:50',
          id: '2381',
          location: 'BA113',
          name: 'Lab L9',
          start: '13:30',
          type: 'LAB'
        }
      ]
    },
    {
      available: 0,
      capacity: 52,
      id: '2435 (28)',
      meetings: [
        {
          available: 195,
          capacity: 330,
          day: 'tuesday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 195,
          capacity: 330,
          day: 'thursday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 0,
          capacity: 52,
          day: 'monday',
          end: '16:50',
          id: '28',
          location: 'BA113',
          name: 'Lab L1',
          start: '14:30',
          type: 'LAB'
        }
      ]
    },
    {
      available: 19,
      capacity: 52,
      id: '2435 (2382)',
      meetings: [
        {
          available: 195,
          capacity: 330,
          day: 'tuesday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 195,
          capacity: 330,
          day: 'thursday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 19,
          capacity: 52,
          day: 'thursday',
          end: '18:20',
          id: '2382',
          location: 'BA113',
          name: 'Lab L10',
          start: '16:0',
          type: 'LAB'
        }
      ]
    },
    {
      available: 38,
      capacity: 52,
      id: '2435 (2521)',
      meetings: [
        {
          available: 195,
          capacity: 330,
          day: 'tuesday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 195,
          capacity: 330,
          day: 'thursday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 38,
          capacity: 52,
          day: 'friday',
          end: '10:50',
          id: '2521',
          location: 'BA113',
          name: 'Lab L11',
          start: '8:30',
          type: 'LAB'
        }
      ]
    },
    {
      available: 40,
      capacity: 52,
      id: '2435 (2522)',
      meetings: [
        {
          available: 195,
          capacity: 330,
          day: 'tuesday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 195,
          capacity: 330,
          day: 'thursday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 40,
          capacity: 52,
          day: 'friday',
          end: '14:20',
          id: '2522',
          location: 'BA113',
          name: 'Lab L12',
          start: '12:0',
          type: 'LAB'
        }
      ]
    },
    {
      available: 30,
      capacity: 52,
      id: '2435 (3688)',
      meetings: [
        {
          available: 195,
          capacity: 330,
          day: 'tuesday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 195,
          capacity: 330,
          day: 'thursday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 30,
          capacity: 52,
          day: 'monday',
          end: '19:20',
          id: '3688',
          location: 'BA113',
          name: 'Lab L13',
          start: '17:0',
          type: 'LAB'
        }
      ]
    },
    {
      available: 7,
      capacity: 30,
      id: '2435 (3728)',
      meetings: [
        {
          available: 195,
          capacity: 330,
          day: 'tuesday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 195,
          capacity: 330,
          day: 'thursday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 7,
          capacity: 30,
          day: 'tuesday',
          end: '21:20',
          id: '3728',
          location: 'N1055',
          name: 'Lab L14',
          start: '19:0',
          type: 'LAB'
        }
      ]
    },
    {
      available: 0,
      capacity: 52,
      id: '2435 (30)',
      meetings: [
        {
          available: 195,
          capacity: 330,
          day: 'tuesday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 195,
          capacity: 330,
          day: 'thursday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 0,
          capacity: 52,
          day: 'tuesday',
          end: '15:20',
          id: '30',
          location: 'BA113',
          name: 'Lab L2',
          start: '13:0',
          type: 'LAB'
        }
      ]
    },
    {
      available: 41,
      capacity: 52,
      id: '2435 (31)',
      meetings: [
        {
          available: 195,
          capacity: 330,
          day: 'tuesday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 195,
          capacity: 330,
          day: 'thursday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 41,
          capacity: 52,
          day: 'tuesday',
          end: '17:50',
          id: '31',
          location: 'BA113',
          name: 'Lab L3',
          start: '15:30',
          type: 'LAB'
        }
      ]
    },
    {
      available: 51,
      capacity: 52,
      id: '2435 (32)',
      meetings: [
        {
          available: 195,
          capacity: 330,
          day: 'tuesday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 195,
          capacity: 330,
          day: 'thursday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 51,
          capacity: 52,
          day: 'wednesday',
          end: '14:50',
          id: '32',
          location: 'BA113',
          name: 'Lab L4',
          start: '12:30',
          type: 'LAB'
        }
      ]
    },
    {
      available: 0,
      capacity: 52,
      id: '2435 (389)',
      meetings: [
        {
          available: 195,
          capacity: 330,
          day: 'tuesday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 195,
          capacity: 330,
          day: 'thursday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 0,
          capacity: 52,
          day: 'wednesday',
          end: '17:20',
          id: '389',
          location: 'BA113',
          name: 'Lab L5',
          start: '15:0',
          type: 'LAB'
        }
      ]
    },
    {
      available: 39,
      capacity: 52,
      id: '2435 (390)',
      meetings: [
        {
          available: 195,
          capacity: 330,
          day: 'tuesday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 195,
          capacity: 330,
          day: 'thursday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 39,
          capacity: 52,
          day: 'wednesday',
          end: '19:50',
          id: '390',
          location: 'BA113',
          name: 'Lab L6',
          start: '17:30',
          type: 'LAB'
        }
      ]
    },
    {
      available: 42,
      capacity: 52,
      id: '2435 (1530)',
      meetings: [
        {
          available: 195,
          capacity: 330,
          day: 'tuesday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 195,
          capacity: 330,
          day: 'thursday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 42,
          capacity: 52,
          day: 'thursday',
          end: '10:50',
          id: '1530',
          location: 'BA113',
          name: 'Lab L7',
          start: '8:30',
          type: 'LAB'
        }
      ]
    },
    {
      available: 41,
      capacity: 52,
      id: '2435 (1655)',
      meetings: [
        {
          available: 195,
          capacity: 330,
          day: 'tuesday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 195,
          capacity: 330,
          day: 'thursday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 41,
          capacity: 52,
          day: 'thursday',
          end: '13:20',
          id: '1655',
          location: 'BA113',
          name: 'Lab L8',
          start: '11:0',
          type: 'LAB'
        }
      ]
    },
    {
      available: 14,
      capacity: 52,
      id: '2435 (2381)',
      meetings: [
        {
          available: 195,
          capacity: 330,
          day: 'tuesday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 195,
          capacity: 330,
          day: 'thursday',
          end: '18:50',
          id: '2435',
          location: 'LH1001',
          name: 'Lec B',
          start: '17:30',
          type: 'LEC'
        },
        {
          available: 14,
          capacity: 52,
          day: 'thursday',
          end: '15:50',
          id: '2381',
          location: 'BA113',
          name: 'Lab L9',
          start: '13:30',
          type: 'LAB'
        }
      ]
    },
    {
      available: 24,
      capacity: 32,
      id: '2935 (2936)',
      meetings: [
        {
          available: 24,
          capacity: 32,
          day: 'tuesday',
          end: '21:50',
          id: '2935',
          location: 'GRH106',
          name: 'Lec BR',
          start: '19:0',
          type: 'LEC'
        },
        {
          available: 24,
          capacity: 32,
          day: 'thursday',
          end: '21:20',
          id: '2936',
          location: 'GRH106',
          name: 'Lab 1A',
          start: '19:0',
          type: 'LAB'
        }
      ]
    }
  ]
}"
`;

exports[`course.wlu.parser parses a course with no sections correctly 1`] = `"false"`;
